src/types/templates.ts

```diff
@@ -0,0 +1,7 @@
+type StringTemplate = string;
+
+type FunctionTemplate = (data: unknown) => string;
+
+export type Templates = {
+  [key: string]: StringTemplate | FunctionTemplate;
```

Comment:
Templates are already typed in a generic way:

https://github.com/algolia/instantsearch.js/blob/c7c53de037959070422c7be85882423078dc1253/src/types/widget.ts#L194-L196

Reply:
ah I missed it. I'll use that one then